tests/e2e/controller_test.go

```diff
@@ -228,27 +212,91 @@ func TestMain(m *testing.M) {
 	flag.BoolVar(&testOpts.operatorControllerTest, "test-operator-controller", true, "run operator controller tests")
 	flag.BoolVar(&testOpts.webhookTest, "test-webhook", true, "run webhook tests")
 
-	componentNames := strings.Join(maps.Keys(componentsTestSuites), ", ")
-	flag.Var(&testOpts.components, "test-component", "run tests for the specified component. valid components names are: "+componentNames)
+	componentNames := strings.Join(testOpts.components.Names(), ", ")
+	flag.BoolVar(&testOpts.components.enabled, "test-components", testOpts.components.enabled, "enable tests for components")
+	flag.Var(&testOpts.components.flags, "test-component", "run tests for the specified component. valid components names are: "+componentNames)
```

Comment:
General thoughts about flags, I see it does `*i = append(*i, value)` in Set. Would it be possible to `strings.Split(value, ",")` to make it possible to set `--test-component=dashboard,!kserve` ?

Reply:
I'll do it in a follow up PR